modules/sr/robot/coordinates/tests.py

```diff
@@ -368,7 +368,9 @@ def test_polar(self) -> None:
 
         for cartesian, expected in cases:
             with self.subTest(cartesian):
-                self.assertEqual(expected, polar_from_cartesian(cartesian))
+                self.assertAlmostEqual(expected.length, polar_from_cartesian(cartesian).length)
+                self.assertAlmostEqual(expected.rot_x, polar_from_cartesian(cartesian).rot_x)
+                self.assertAlmostEqual(expected.rot_y, polar_from_cartesian(cartesian).rot_y)
```

Comment:
If we really need to do this please could we avoid the duplicate calls to `polar_from_cartesian`?
However unfortunately this is a semantic change versus the original assertion and one which is not very future-proof.
Suggest that instead of this we perhaps force the rounding ourselves?
```suggestion
                actual = polar_from_cartesian(cartesian)
                # Cope with floating point differences
                rounded = PolarCoord(*(round(x, 7) for x in actual))

                self.assertEqual(expected, actual)
```

Reply:
87bd35c2aa9c3c0e502865d016b27f1251604d0c